python-3.11/gh-61460.patch

```diff
@@ -0,0 +1,473 @@
+ 
+ def answer_challenge(connection, authkey):
++    """Respond to HMAC authentication challenge (client side).
++
++    Receives a challenge from the server and sends back an HMAC response.
++    Detects the server's protocol version by challenge length.
++
++    Protocol:
++      1. Client receives: CHALLENGE prefix + random bytes
++      2. Client detects algorithm by challenge length:
++         - 20 bytes: Legacy server, compute MD5 HMAC
++         - 32 bytes: Modern server, compute SHA256 HMAC
++      3. Client sends HMAC response
++      4. Client receives WELCOME on success or FAILURE on error
++
++    Args:
++        connection: Connection object with send_bytes/recv_bytes methods
++        authkey: Shared secret key (bytes)
++
++    Raises:
++        ValueError: If authkey is not bytes
++        AuthenticationError: If challenge format invalid or server rejects response
++    """
+     import hmac
+     if not isinstance(authkey, bytes):
+         raise ValueError(
+             "Authkey must be bytes, not {0!s}".format(type(authkey)))
+     message = connection.recv_bytes(256)         # reject large message
+-    assert message[:len(CHALLENGE)] == CHALLENGE, 'message = %r' % message
++
++    if not message.startswith(CHALLENGE):
++        raise AuthenticationError(
++            'message missing challenge prefix (expected {0!r})'.format(CHALLENGE))
++
+     message = message[len(CHALLENGE):]
+-    digest = hmac.new(authkey, message, 'md5').digest()
++
++    # Detect protocol based on challenge length
++    if len(message) == MESSAGE_LENGTH:
++        # Legacy 20-byte challenge: server expects MD5
++        try:
++            digest = hmac.new(authkey, message, 'md5').digest()
++        except ValueError:
++            raise AuthenticationError(
++                'legacy server requires MD5 but MD5 not available (FIPS mode?)')
++    elif len(message) == MESSAGE_LENGTH_SHA256:
++        # Modern 32-byte challenge: server expects SHA256
++        try:
++            digest = hmac.new(authkey, message, 'sha256').digest()
++        except ValueError:
++            raise AuthenticationError(
++                'modern server requires SHA256 but SHA256 not available')
++    else:
++        raise AuthenticationError(
++            'unexpected challenge length {0} (expected {1} or {2})'.format(
++                len(message), MESSAGE_LENGTH, MESSAGE_LENGTH_SHA256))
++
+     connection.send_bytes(digest)
+     response = connection.recv_bytes(256)        # reject large message
+     if response != WELCOME:
+-        raise AuthenticationError('digest sent was rejected')
++        raise AuthenticationError('server rejected authentication')
+ 
+ #
+ # Support for using xmlrpclib for serialization
+diff --git a/Lib/test/_test_multiprocessing.py b/Lib/test/_test_multiprocessing.py
+index e096401c4cf..40cec1057bb 100644
+--- a/Lib/test/_test_multiprocessing.py
++++ b/Lib/test/_test_multiprocessing.py
```

Comment:
Are these tests executed anywhere? (Build time, or package test time, or image test time, or in openssl-config-fipshardened package tests)?